### PR TITLE
Add LIV SDK

### PR DESCRIPTION
**Pull request by twitchplayskh**
ported over SteffanDonal liv api support to work with open brush so people can use LIV with open brush lets you add a 3d avatar camera for vr

notes marker

### CODE REVIEW

**Comment by billyquith** [comment]
Hi. Is this optional/conditional? For dev, fork, or given platform, you might not want to include this.

**Comment by SteffanDonal** [comment]
LIV is completely nonactive until LIV itself attempts to capture the process (or Unity editor, it works very well there)
There are also defines for excluding LIV from non-PCVR builds IIRC

**Comment by billyquith** [comment]
> LIV is completely nonactive until LIV itself attempts to capture the process (or Unity editor, it works very well there)
> There are also defines for excluding LIV from non-PCVR builds IIRC

Inactive, but still compiled into the app?

There is a conditional for `LIV_UNIVERSAL_RENDER` i.e. choice of render path. How would I remove this from compilation altogether if I didn't want this feature on a platform? E.g. platform performance issues/unsupported.

Which platforms is it tested on please? Any mobile platforms?

**Comment by TimAidley** [comment]
A couple of issues:
  Firstly, if possible it should be moved under the Assets/ThirdParty/ folder.

Secondly, there appears to be no license included - what license is the code distributed under? There should be a LICENSE file or similar included with it.

**Comment by twitchplayskh** [comment]
> 
> 
> > LIV is completely nonactive until LIV itself attempts to capture the process (or Unity editor, it works very well there)
> > There are also defines for excluding LIV from non-PCVR builds IIRC
> 
> Inactive, but still compiled into the app?
> 
> There is a conditional for `LIV_UNIVERSAL_RENDER` i.e. choice of render path. How would I remove this from compilation altogether if I didn't want this feature on a platform? E.g. platform performance issues/unsupported.
> 
> Which platforms is it tested on please? Any mobile platforms?

you need the pc app to run it it should not mees with things when its not used

**Comment by twitchplayskh** [comment]
> 
> 
> A couple of issues:
> Firstly, if possible it should be moved under the Assets/ThirdParty/ folder.
> 
> Secondly, there appears to be no license included - what license is the code distributed under? There should be a LICENSE file or similar included with it.

ok i will fix this thanks

**Comment by SteffanDonal** [comment]
> > LIV is completely nonactive until LIV itself attempts to capture the process (or Unity editor, it works very well there)
> > There are also defines for excluding LIV from non-PCVR builds IIRC
> 
> Inactive, but still compiled into the app?
> 
> There is a conditional for `LIV_UNIVERSAL_RENDER` i.e. choice of render path. How would I remove this from compilation altogether if I didn't want this feature on a platform? E.g. platform performance issues/unsupported.
> 
> Which platforms is it tested on please? Any mobile platforms?

Entirely inactive, still compiled into the app. See:
https://github.com/icosa-gallery/open-brush/blob/2094339be6851731d293f2574c1fc706ee388d84/Assets/LIV/Scripts/SDKBridge.cs#L10

The entire LIV implementation is gated around the `GetIsCaptureActive` boolean. There are no code paths that result in anything more than a bool check unless you expressly have LIV running and intend to capture.  
On non `(UNITY_STANDALONE_WIN || UNITY_EDITOR_WIN) && UNITY_64` built targets, this is hard-wired to return false.

It would be fairly trivial to expand this to outright exclude any code whatsoever, I'm happy to make that change and push it to LIV's fork in the interests of interoperability. 



> A couple of issues:
> Firstly, if possible it should be moved under the Assets/ThirdParty/ folder.
> 
> Secondly, there appears to be no license included - what license is the code distributed under? There should be a LICENSE file or similar included with it.

Apologies, this should be included with LIV's SDK. This code is MIT licensed.

**Comment by twitchplayskh** [comment]
> 
> 
> > > LIV is completely nonactive until LIV itself attempts to capture the process (or Unity editor, it works very well there)
> > > There are also defines for excluding LIV from non-PCVR builds IIRC
> > 
> > 
> > Inactive, but still compiled into the app?
> > There is a conditional for `LIV_UNIVERSAL_RENDER` i.e. choice of render path. How would I remove this from compilation altogether if I didn't want this feature on a platform? E.g. platform performance issues/unsupported.
> > Which platforms is it tested on please? Any mobile platforms?
> 
> Entirely inactive, still compiled into the app. See:
> 
> https://github.com/icosa-gallery/open-brush/blob/2094339be6851731d293f2574c1fc706ee388d84/Assets/LIV/Scripts/SDKBridge.cs#L10
> 
> The entire LIV implementation is gated around the `GetIsCaptureActive` boolean. There are no code paths that result in anything more than a bool check unless you expressly have LIV running and intend to capture.
> On non `(UNITY_STANDALONE_WIN || UNITY_EDITOR_WIN) && UNITY_64` built targets, this is hard-wired to return false.
> 
> It would be fairly trivial to expand this to outright exclude any code whatsoever, I'm happy to make that change and push it to LIV's fork in the interests of interoperability.
> 
> > A couple of issues:
> > Firstly, if possible it should be moved under the Assets/ThirdParty/ folder.
> > Secondly, there appears to be no license included - what license is the code distributed under? There should be a LICENSE file or similar included with it.
> 
> Apologies, this should be included with LIV's SDK. This code is MIT licensed.

oh ok when i have all the files i will update my pull to fix all that i need to fix

File: ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -17,7 +17,7 @@ PlayerSettings:
   defaultCursor: {fileID: 0}
   cursorHotspot: {x: 0, y: 0}
   m_SplashScreenBackgroundColor: {r: 0.13725491, g: 0.12156863, b: 0.1254902, a: 1}
-  m_ShowUnitySplashScreen: 0
+  m_ShowUnitySplashScreen: 1
```
[the remark this answers is not in the document]
Comment (mikeskydev) [reply ProjectSettings/ProjectSettings.asset:20]: I think it would automatically reset, but ideally we should keep it as 0 in the repository

Comment (mikeskydev) [inline ProjectSettings/ProjectSettings.asset:20]: Change back to `0`

**Review by mikeskydev — CHANGES_REQUESTED** [review]
Happy to include LIV as it's used quite a lot by creators, and current support in TB leaves a lot to be desired. Please action the following changes:
- Move LIV directory to `Assets/ThirdParty`
- Change back splash screen
- Add the MIT licence mentioned above at `Assets/ThirdParty/LIV/LICENCE` (The MIT licence is a standard and you could add this yourself, with credit to LIV or however @SteffanDonal wishes it to be shown.

File: ProjectSettings/ProjectSettings.asset (hunk 1 above)
Comment (twitchplayskh) [reply ProjectSettings/ProjectSettings.asset:20]: i will do this next week as i do not have time this week

Comment (mikeskydev) [reply ProjectSettings/ProjectSettings.asset:20]: Okay! and yes it is

**Comment by mikeskydev** [comment]
Closing in favour of #57

**Pull request closed without merging**